server/src/main/java/com/objectcomputing/checkins/services/pulseresponse/PulseResponseServicesImpl.java

```diff
@@ -2,81 +2,123 @@
 
 import com.objectcomputing.checkins.exceptions.BadArgException;
 import com.objectcomputing.checkins.services.memberprofile.MemberProfileRepository;
+import com.objectcomputing.checkins.services.memberprofile.MemberProfileServices;
+import com.objectcomputing.checkins.services.memberprofile.currentuser.CurrentUserServices;
+import com.objectcomputing.checkins.services.permissions.Permission;
+import com.objectcomputing.checkins.services.role.role_permissions.RolePermissionServices;
 import jakarta.inject.Singleton;
 import jakarta.validation.constraints.NotNull;
 
 import java.time.LocalDate;
-import java.util.HashSet;
 import java.util.Set;
 import java.util.UUID;
+import java.util.stream.Collectors;
 
 @Singleton
 public class PulseResponseServicesImpl implements PulseResponseService {
 
     private final PulseResponseRepository pulseResponseRepo;
+    private final MemberProfileServices memberProfileServices;
     private final MemberProfileRepository memberRepo;
+    private final CurrentUserServices currentUserServices;
+    private final RolePermissionServices rolePermissionServices;
 
-    public PulseResponseServicesImpl(PulseResponseRepository pulseResponseRepo,
-                                     MemberProfileRepository memberRepo) {
+    public PulseResponseServicesImpl(
+            PulseResponseRepository pulseResponseRepo,
+            MemberProfileServices memberProfileServices,
+            MemberProfileRepository memberRepo,
+            CurrentUserServices currentUserServices,
+            RolePermissionServices rolePermissionServices
+    ) {
         this.pulseResponseRepo = pulseResponseRepo;
+        this.memberProfileServices = memberProfileServices;
         this.memberRepo = memberRepo;
+        this.currentUserServices = currentUserServices;
+        this.rolePermissionServices = rolePermissionServices;
     }
-    
+
     @Override
     public PulseResponse save(PulseResponse pulseResponse) {
+        UUID currentUserId = currentUserServices.getCurrentUser().getId();
         PulseResponse pulseResponseRet = null;
-        if(pulseResponse!=null){
+        if (pulseResponse != null) {
             final UUID memberId = pulseResponse.getTeamMemberId();
```

Comment:
Given my comment about only being able to create/update your own responses. It might be worthwhile to turn the `teammemberid` field into one that is annotated with and populated via `@CreatedBy`. Thoughts?

Reply:
I had a look, but I am unsure as to how to get the test working with `@CreatedBy` 🤔 

Should I change

```
            } else if (!currentUserId.equals(memberId) && !isSubordinateTo(memberId, currentUserId)) {
```

in save and update to

```
            } else if (!currentUserId.equals(memberId)) {
```

To more closely match your comment?  We can then raise a card to investigate `@CreatedBy` and I can pick it up when I return on the 3rd?

Reply:
That works, yes. Sorry for the slow reply. I missed yours completely.